cmd/test-server/main.go

```diff
@@ -143,41 +128,271 @@ func echoHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func getTcpRttStats(arr []float64) (float64, float64, float64, float64) {
-	data := stats.LoadRawData(arr)
-	min, _ := stats.Min(data)
-	avg, _ := stats.Mean(data)
-	max, _ := stats.Max(data)
-	stddev, _ := stats.StandardDeviation(data)
-	return min, avg, max, stddev
+// Parse IP and TCP headers from ICMP Response Payload
+func getHeadersFromICMPResponsePayload(data []byte) (*layers.IPv4, *layers.TCP, error) {
+	if len(data) < 1 {
+		return nil, nil, errors.New("Invalid IP header")
+	}
+	ipHeaderLength := int((data[0] & 0x0F) * 4)
+
+	if len(data) < ipHeaderLength {
+		return nil, nil, errors.New("Length of received ICMP packet too short to decode IP")
+	}
+	ip := layers.IPv4{}
+	tcp := layers.TCP{}
+	ipErr := ip.DecodeFromBytes(data[0:], gopacket.NilDecodeFeedback)
+	tcpErr := tcp.DecodeFromBytes(data[ipHeaderLength:], gopacket.NilDecodeFeedback)
+
+	if ipErr != nil && tcpErr != nil {
+		return nil, nil, ipErr
+	}
+
+	return &ip, &tcp, nil
+}
+
+// Returns current time, used to reset counter timer
+func resetTimerForCounter() time.Time {
+	return time.Now()
```

Comment:
I find it useful to always use UTC time. Makes it easier to deal with times if we never have to worry about time zones.
```suggestion
	return time.Now().UTC()
```